### PR TITLE
Require soft deletes for searchable snapshots

### PR DESCRIPTION
**Pull request by DaveCTurner**
A simpler fix that lets us support deleted docs in searchable snapshots.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java [hunk 1]
```diff
@@ -71,13 +72,12 @@ public void testCreateAndRestoreSearchableSnapshot() throws Exception {
                 .setSettings(Settings.builder().put("location", repo).put("chunk_size", randomIntBetween(100, 1000), ByteSizeUnit.BYTES))
         );
 
-        createIndex(indexName);
+        assertAcked(prepareCreate(indexName, Settings.builder().put(INDEX_SOFT_DELETES_SETTING.getKey(), true)));
```
Comment (DaveCTurner) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:75]: In fact this is always true in `master`, but not in `7.x`. Setting it in both places for ease of backport.

**Comment by ywelsch** [comment]
I would rather like to avoid this, as it won't allow us to load most indices created with 6.x.

Can we just adapt the test change for now with a new TODO?

**Comment by DaveCTurner** [comment]
Do you mean to revert the change to `SearchableSnapshotDirectory` and not throw an IAE if soft deletes is disabled? Yes, we can do that, the main drawback is that if you try and mount a 6.x snapshot that contains deletes then you will get mysterious shard failures.

I expect we'll be using Lucene 8.6 and therefore be able to fix this properly before too long.

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/1

**Review by ywelsch — APPROVED** [review]

**Pull request merged**